### PR TITLE
fix(icon-service): destroy parent container on HMR

### LINKED ISSUE #9228: addSVGIcon does not cache the icons already added.
## Description  
The icons are not cached when added to a component via addSVGIcon() method. This causes adding svgs that already exist to the DOM.

 * igniteui-angular version: N/A
 * browser: N/A

## Steps to reproduce  

1. Load [this sample](https://stackblitz.com/edit/github-dmmnge?file=src%2Fapp%2Fapp.component.html) 
2. Check the items in the svg container > they should be 7
3. Make any change, e.g. add a new line in app.component.ts, save and check the items again

## Result  
The svg container has 14 items.

## Expected result  
They should still be 7.

### PR DESCRIPTION
**Pull request by simeonoff**
Closes #9228  

### Additional information (check all that apply):
 - [x] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [ ] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/icon/icon.service.spec.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { DOCUMENT } from '@angular/common';
 
 import { configureTestSuite } from '../test-utils/configure-suite';
 import { first } from 'rxjs/operators';
+import { HttpClientModule } from '@angular/common/http';
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/icon/icon.service.spec.ts:7]: Use `HttpClientTestingModule`

File: projects/igniteui-angular/src/lib/icon/icon.service.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
-import { Injectable, SecurityContext, Inject } from '@angular/core';
+import { Injectable, SecurityContext, Inject, OnDestroy, Optional } from '@angular/core';
 import { DomSanitizer } from '@angular/platform-browser';
 import { DOCUMENT } from '@angular/common';
+import { HttpClient } from '@angular/common/http';
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/icon/icon.service.ts:4]: There was an issue with importing `HttpClient` in the icon module in the past. Can you check what the issue was?

**Review by kdinev — CHANGES_REQUESTED** [review]

File: projects/igniteui-angular/src/lib/icon/icon.service.ts (hunk 2 above)
Comment (rkaraivanov) [reply projects/igniteui-angular/src/lib/icon/icon.service.ts:4]: I recall there were problems with http interceptors in the past. Especially if our icon module was lazy loaded.

Comment (simeonoff) [reply projects/igniteui-angular/src/lib/icon/icon.service.ts:4]: Reading the issue, it seems it only happens in JIT compiled apps?

Comment (kdinev) [reply projects/igniteui-angular/src/lib/icon/icon.service.ts:4]: @rkaraivanov is right. It was something with HttpInterceptors, but honestly I don't remember what the issue was.

**Comment by rkaraivanov** [comment]
@simeonoff 

If you still want to keep the observable behavior for fetching the icons
you can use the [ajax](https://rxjs-dev.firebaseapp.com/api/ajax/ajax) operator

**Review by kdinev — APPROVED** [review]

**Comment by simeonoff** [comment]
> @simeonoff
> 
> If you still want to keep the observable behavior for fetching the icons
> you can use the [ajax](https://rxjs-dev.firebaseapp.com/api/ajax/ajax) operator

Thanks. It's fine the way it was in this case.

**Review by kdinev — APPROVED** [review]

**Review by kdinev — APPROVED** [review]

**Pull request merged**